### ISSUE TITLE
Reference to database and/or server name in 'master.sys.schemas' is not supported in this version of SQL Server.

### ISSUE THREAD

**Issue by lukaseder** [jOOQ/jOOQ#12291]
When doing the following on an Azure SQL database:

```java
Meta meta = ctx.meta();

Catalog catalog = meta.getCatalog("test");
Schema schema = catalog.getSchema("dbo");

Table<?> table = schema.getTable("t_book");
List<? extends ForeignKey<?, ?>> fks = table.getReferences();
```

Then there's an exception like this:

```
Exception in thread "main" org.jooq.exception.DataAccessException: Error while running ConnectionCallable
	at org.jooq.impl.DefaultDSLContext.connectionResult(DefaultDSLContext.java:662)
	at org.jooq.impl.MetaImpl.meta(MetaImpl.java:187)
	at org.jooq.impl.MetaImpl$MetaCatalog.getSchemas(MetaImpl.java:284)
	at org.jooq.impl.MetaImpl.lambda$2(MetaImpl.java:242)
	at org.jooq.impl.Tools.flatMap(Tools.java:2107)
	at org.jooq.impl.MetaImpl.getSchemas0(MetaImpl.java:242)
	at org.jooq.impl.AbstractMeta.lambda$1(AbstractMeta.java:152)
	at org.jooq.impl.AbstractMeta.get(AbstractMeta.java:371)
	at org.jooq.impl.AbstractMeta.initSchemas(AbstractMeta.java:152)
	at org.jooq.impl.AbstractMeta.getSchemas(AbstractMeta.java:144)
	at org.jooq.impl.MetaImpl$MetaTable.getReferences(MetaImpl.java:824)
	at org.jooq.testscripts.JDBC.main(JDBC.java:45)
Caused by: com.microsoft.sqlserver.jdbc.SQLServerException: Reference to database and/or server name in 'master.sys.schemas' is not supported in this version of SQL Server.
	at com.microsoft.sqlserver.jdbc.SQLServerException.makeFromDatabaseError(SQLServerException.java:262)
	at com.microsoft.sqlserver.jdbc.SQLServerStatement.getNextResult(SQLServerStatement.java:1632)
	at com.microsoft.sqlserver.jdbc.SQLServerPreparedStatement.doExecutePreparedStatement(SQLServerPreparedStatement.java:602)
	at com.microsoft.sqlserver.jdbc.SQLServerPreparedStatement$PrepStmtExecCmd.doExecute(SQLServerPreparedStatement.java:524)
	at com.microsoft.sqlserver.jdbc.TDSCommand.execute(IOBuffer.java:7418)
	at com.microsoft.sqlserver.jdbc.SQLServerConnection.executeCommand(SQLServerConnection.java:3272)
	at com.microsoft.sqlserver.jdbc.SQLServerStatement.executeCommand(SQLServerStatement.java:247)
	at com.microsoft.sqlserver.jdbc.SQLServerStatement.executeStatement(SQLServerStatement.java:222)
	at com.microsoft.sqlserver.jdbc.SQLServerPreparedStatement.executeQueryInternal(SQLServerPreparedStatement.java:460)
	at com.microsoft.sqlserver.jdbc.SQLServerDatabaseMetaData.getSchemasInternal(SQLServerDatabaseMetaData.java:1498)
	at com.microsoft.sqlserver.jdbc.SQLServerDatabaseMetaData.getSchemas(SQLServerDatabaseMetaData.java:1508)
	at org.jooq.impl.MetaImpl$MetaCatalog.lambda$0(MetaImpl.java:288)
	at org.jooq.impl.MetaImpl.lambda$0(MetaImpl.java:187)
	at org.jooq.impl.DefaultDSLContext.connectionResult(DefaultDSLContext.java:656)
	... 11 more
```

**Comment by lukaseder** [comment]
The problem is that our `DatabaseMetaData` backed `MetaImpl` tries to fetch all the catalogs and their schemas, when this isn't possible on SQL Azure, even though this call is possible: `connection.getMetaData().getCatalogs()`

**Comment by lukaseder** [comment]
Fixed in jOOQ 3.16.0, 3.15.2 (https://github.com/jOOQ/jOOQ/issues/12292), and 3.14.14 (https://github.com/jOOQ/jOOQ/issues/12293)
